client-js/app/elements/labrad-plot.ts

```diff
@@ -181,6 +188,32 @@ export class Plot extends polymer.Base {
     }
   }
 
+  @observe('drawMode2D')
+  drawMode2DChanged(newMode: string, oldMode: string) {
+    switch (newMode) {
+      case 'dots':
+        this.$.dots.style.color = 'black';
+        this.$.rectfill.style.color = '#AAAAAA';
+        this.$.vargrid.style.color = '#AAAAAA';
+        break;
+
+      case 'rectfill':
+        this.$.dots.style.color = '#AAAAAA';
+        this.$.rectfill.style.color = 'black';
+        this.$.vargrid.style.color = '#AAAAAA';
+        break;
+
+      case 'vargrid':
+        this.$.dots.style.color = '#AAAAAA';
+        this.$.rectfill.style.color = '#AAAAAA';
+        this.$.vargrid.style.color = 'black';
+        break;
+    }
+    if (oldMode && newMode !== oldMode) {
```

Comment:
This triggers a redraw when the mode changes. The check on whether `oldMode` itself is non-null is because this observer gets fired when the property is first set, but in that case we don't need to redraw (the mode wasn't changed _from_ a previous mode).